pkg/types/baremetal/platform.go

```diff
@@ -61,4 +61,19 @@ type Platform struct {
 
 	// DNSVIP is the VIP to use for internal DNS communication
 	DNSVIP string `json:"dnsVIP"`
+
+	// ProvisioningInterface is the network interface used to provision new hosts.
+	ProvisioningInterface string `json:"provisioningInterface"`
+
+	// ProvisioningNetworkCIDR defines the network to use for provisioning.
+	ProvisioningNetworkCIDR string `json:"provisioningNetworkCIDR"`
+
+	// ProvisioningIP is the IP address to assign to the provisioning interface.
+	ProvisioningIP string `json:"provisioningIP"`
+
+	// ProvisioningDHCPRange is the DHCP range to use to assign hosts during provisioning.
+	ProvisioningDHCPRange string `json:"provisioningDHCPRange"`
+
+	// RhcosImageURL is the URL used to fetch the image for provisioning new hosts.
+	RhcosImageURL string `json:"rhcosImageURL"`
```

Comment:
as above I don't think we should add the rhcos url here, instead we'll reference the platform.Image